lib/header.c

```diff
@@ -394,10 +394,16 @@ static inline int strtaglen(const char *str, rpm_count_t c, const char *end)
     const char *s;
 
     if (end) {
-	if (str >= end)
+	if (start >= end)
```

Comment:
Hmm, doesn't the newly added condition below make this check entirely redundant?

Reply:
Yes and no.  It wasn’t redundant before #1496, but it definitely is now.  Deleted.